runtime/container_linux.go

```diff
@@ -284,6 +284,52 @@ func (c *container) Stats() (*Stat, error) {
 	}, nil
 }
 
+// Status implements the runtime Container interface.
+func (c *container) Status() (State, error) {
+	args := c.runtimeArgs
+	args = append(args, "state", c.id)
+
+	out, err := exec.Command(c.runtime, args...).CombinedOutput()
+
+	// If only error happens, it is useful to return the detail.
+	if err != nil {
+		return "", fmt.Errorf(string(out))
+	}
+
+	// We may not know the exactly structure of the response if we
+	// have more runtimes other than OCI/runc.
+	var f interface{}
```

Comment:
I don't like unmarshaling into an interface at all.  You are better off creating a type and using that.

You are better off writing it like this:

```go
// Status implements the runtime Container interface.
func (c *container) Status() (State, error) {
	args := c.runtimeArgs
	args = append(args, "state", c.id)

	out, err := exec.Command(c.runtime, args...).CombinedOutput()
	if err != nil {
		return "", fmt.Errorf(string(out))
	}

	// We may not know the exactly structure of the response if we
	// have more runtimes other than OCI/runc.
	var s struct {
            Status string `json:"status"`
        }
	if err := json.Unmarshal(out, &s); err != nil {
		return "", err
	}
        return s.Status, nil
}
```